mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/component/value/DefaultPoolingProfileValueDeclarer.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.internal.component.value;
+
+import static org.mule.runtime.api.config.PoolingProfile.DEFAULT_EVICTION_INTERVAL_MILLIS;
+import static org.mule.runtime.api.config.PoolingProfile.DEFAULT_MAX_POOL_ACTIVE;
+import static org.mule.runtime.api.config.PoolingProfile.DEFAULT_MAX_POOL_IDLE;
+import static org.mule.runtime.api.config.PoolingProfile.DEFAULT_MAX_POOL_WAIT;
+import static org.mule.runtime.api.config.PoolingProfile.DEFAULT_MIN_EVICTION_MILLIS;
+import static org.mule.runtime.extension.api.component.value.PoolingProfileValueDeclarer.ExhaustedAction.WHEN_EXHAUSTED_GROW;
+import static org.mule.runtime.extension.api.component.value.PoolingProfileValueDeclarer.InitialisationPolicy.INITIALISE_ONE;
+
+import org.mule.runtime.api.config.PoolingProfile;
+import org.mule.runtime.extension.api.component.value.PoolingProfileValueDeclarer;
+
+/**
+ * Default implementation of {@link PoolingProfileValueDeclarer}
+ *
+ * @since 1.5
+ */
+class DefaultPoolingProfileValueDeclarer implements PoolingProfileValueDeclarer, DefaultValueDeclarer.HasValue {
```

Comment:
implementation should be in mule

Reply:
Should I also move the implementation of the existing ValueDeclarers and the ComponentParameterization implementation?